src/NHibernate/Bytecode/Lightweight/Getter.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using NHibernate.Properties;
+
+namespace NHibernate.Bytecode.Lightweight
+{
+	/// <summary>
+	/// Contains the <see cref="IGetter"/> instance with an optional optimized delegate.
+	/// </summary>
+	public class Getter
```

Comment:
Why not hiding the fact it is optimized or not?

It could be something like:

```c#
public class OptimizedGetter
{
	private readonly Func<object, object> _getter;

	public OptimizedGetter(IGetter @default) : this(@default, null)
	{
	}

	public OptimizedGetter(IGetter @default, Func<object, object> optimized)
	{
		_getter = optimized ?? @default.Get;
	}

	public object Get(object target)
	{
		return _getter(target);
	}
}
```

It would allow a more straightforward implementation of the access optimizer.

Of course the same should then be done for setters.

Reply:
> Why not hiding the fact it is optimized or not?

I agree, I think it is even better to use a delegate to be consistent with [Get/Set]PropertyValues, which is what I've done. If you think that it would be better to have a class I will change to it.